Java/Algorithms/Min_Max_Algorithm_implementation.java

```diff
@@ -0,0 +1,71 @@
+import java.util.Scanner;
+
+public class Min_Max_Algorithm_implementation {
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter the size of array:-");
+        
+        int n = sc.nextInt();
+        int arr[] = new int[n];
+        
+        System.out.println("Enter elements of array");
+        
+        for(int i = 0; i<n; i++){
+            arr[i] = sc.nextInt();
+        }
+        
+        int low = 0; //Least index of an array/subarrays
+        int high = n - 1; //Highest index of an array/subarrays
+        
+        int output[] = MinMax(arr, low, high);
+        System.out.println("Output:");
+        System.out.println(String.format("Min:- %d Max:- %d", output[0], output[1]));
+    }
+
+    /* 
+        Function that returns minimum and maximum out of 2 elements.
+        It returns an array with first element as minimum and second element as maximum out of 2 elements.
+    */ 
+
+    public static int[] min_max_of_2_elements(int x, int y){
+        int min_max[] = new int[2];
+        if(x < y){
+            min_max[0] = x;
+            min_max[1] = y;
+        }
+        else{
+            min_max[0] = y;
+            min_max[1] = x;
+        }
+        return min_max;
+    }
+
+    public static int[] MinMax(int arr[], int low, int high){
+        // If after dividing array, we have one element left or two elements then return min or max of those elements
+        
+        if(low == high || high == low + 1){
+            return min_max_of_2_elements(arr[low], arr[high]);
+        }
+        
+        else{
+            // Dividing array recursively into smaller parts. 
+            int mid = (low + high)/2;
```

Comment:
@vrandachag Please avoid this technique to find the mid element since this can cause an integer overflow. Use this `int mid = low + ((high - low) / 2);`.

Reply:
Actually, `low` and `high` are indexes of the array, but I agree that there could be a possibility of an **integer overflow** when the array size is **too big**. Should I make the changes? @iamrajiv

Reply:
@Devansu-Yadav Yes please do.

Reply:
@iamrajiv Done.